lib/graphql/types/iso_8601_date_time.rb

```diff
@@ -29,12 +29,13 @@ def self.time_precision=(value)
         @time_precision = value
       end
 
-      # @param value [DateTime]
+      # @param value [Date,DateTime,String]
       # @return [String]
       def self.coerce_result(value, _ctx)
-        value.iso8601(time_precision)
+        value = value.iso8601(time_precision) if value.class == DateTime
```

Comment:
Why use `.class ==` here, instead of `.is_a?(DateTime)`? Are you trying to exclude subclasses of `DateTime`? 

(This might be a breaking change, since this method _used_ to accept anything that responded to `.iso8601(time_precision)`)

Reply:
> Why use `.class ==` here, instead of `.is_a?(DateTime)`? Are you trying to exclude subclasses of `DateTime`?

Call it a less-than-ideal understanding of Ruby semantics. :)

> (This might be a breaking change, since this method used to accept anything that responded to `.iso8601(time_precision)`)

The main reason I didn't do this is that `Date#iso8601` raises an exception when passed `time_precision`, and ignoring the precision leads to broken tests. Can you think of a better way to get around this constraint?

Reply:
How about: 

```ruby 
def coerce_result(value, _ctx)
  case value 
  when DateTime 
    value.iso8601(time_precision)
  when Date 
    DateTime.parse(value.to_s).iso8601(time_precision)
  when String 
    DateTime.parse(value).iso8601(time_precision)
  else 
    # In case some other API-compliant thing is given: 
    value.iso8601(time_precision)
  end 
rescue ArgumentError, NoMethodError 
  raise GraphQL::Error, "An incompatible object (#{value.class}) was given to #{self}. Make sure that only Dates, DateTimes, and well-formatted Strings are used with this type."
end 
```

?

Reply:
Sounds good. I'll incorporate this. Thanks.

Reply:
@rmosolgo Got stuck for a while getting the `when String` case to pass. Turns out `String` was referencing `GraphQL::Types::String` instead of `::String`. Should be good to go now.

Reply:
😖 good catch!